red-pill-contracts-py/README.md

```diff
@@ -0,0 +1,100 @@
+# Detecting Red-Pill Contracts
+
+## Description
+
+Implementation for the detection techniques described in the [report about smart contract evasion techniques][report-web3-evasion] by the [Forta TRi][forta-threat-research-initiative].
+
+Here, "evasion" refers to any tactic that deceives end-users or circumvents defense mechanisms.
+
+More specifically, the bot is focused on contrats that try to detect / evade simulation environments.
+As in the movie "Matrix" they are aware that they live in a simulation, hence the name "red-pill".
+
+## Support
+
+The bots use the transaction traces, so they only runs on Ethereum for now.
+
+## Table of Contents
+
+- [Alerts](#alerts)
+- [Options](#options)
+- [Implementations](#implementations)
+- [Development](#development)
+  - [Changelog](#changelog)
+  - [Todo](#todo)
+  - [Performances](#performances)
+- [Credits](#credits)
+- [License](#license)
+
+## Alerts
+
+The red-pill contracts are spotted when created to perform static analysis on the bytecode:
+
+- `LOGIC-BOMB-RED-PILL-DEPLOYMENT`:
+    - the address of the contract is attached as a label
+
+For all the alerts:
+
+- Type is always set to `Suspicious`
+- Severity is always `Info`
+- Metadata:
+  - `confidence`: the estimated probability of a given detection
+  - `chain_id`: the chain id
+  - `from`: the transaction sender
+  - `to`: the transaction recipient
+  - `anomaly_score`: the alert rate for this combination of bot / alert type
+
+## Options
+
+The bot settings are located in `src/options.py`:
+
+```python
+MIN_CONFIDENCE = 0.7 # probability threshold
+ALERT_HISTORY_SIZE = 16384 # in number of transactions recorded
```

Comment:
in number of alerts?

Reply:
Nope transactions :D
The history has an empty record for transactions without alerts in order to calculate the "alert-rate"